### ISSUE TITLE
Should events not be re-entrant ? 

### ISSUE THREAD

**Issue by Carreau** [napari/napari#6234]
See #6223

In general events and triggering computation on changes seem like it makes it easy to shoot oneself in the foot and have cycle. 

Should we prevented  events in general to be rentrant, and that while processing event X, X cannot be re-emmited again 

## Motivation

Event are great to synchronize values across napari, though it make it hard to figure out which code is going to be triggered when and tend to create loops. I believe a way to properly reason about which code can trigger what would be useful

### Alternative:

Is there any other framework or declarative practice that would allow us to have easier synchronisation of values ?

**Comment by Czaki** [comment]
> Should we prevented events in general to be rentrant, and that while processing event X, X cannot be re-emmited again

Let's think about delaying re emitting until processing is done. We should not block it. Or if we block, then it should raise an exception to prevent cyclic dependencies. Otherwise, we may end up in an inconsistent state.

**Comment by Carreau** [comment]
Yeah, I was thinking or either 1) blocking if re-entring or 2) doing nothing and emit a waring.  Definitely not blocking.
